module/spring-boot-amqp/src/test/java/org/springframework/boot/amqp/autoconfigure/metrics/RabbitMetricsAutoConfigurationTests.java

```diff
@@ -35,14 +36,17 @@
  * Tests for {@link RabbitMetricsAutoConfiguration}.
  *
  * @author Stephane Nicoll
+ * @author Eddú Meléndez
  */
 class RabbitMetricsAutoConfigurationTests {
 
 	private final ApplicationContextRunner contextRunner = new ApplicationContextRunner()
 		.withBean(SimpleMeterRegistry.class)
 		.withConfiguration(AutoConfigurations.of(RabbitAutoConfiguration.class, RabbitMetricsAutoConfiguration.class,
 				MetricsAutoConfiguration.class))
-		.withPropertyValues("management.metrics.use-global-registry=false");
+		.withPropertyValues("management.metrics.use-global-registry=false")
+		.withClassLoader(
+				new FilteredClassLoader("com.rabbitmq.client.amqp", "org.springframework.amqp.rabbitmq.client"));
```

Comment:
I think if you make a `RabbitAutoConfiguration` conditional on bean from our new AMQP 1.0 auto-configuration, then we would not need this kind of classloader filtering.
Since we just don't include that `RabbitAmqpAutoConfiguration` into these existing tests.